### ISSUE TITLE
`no_empty_block` not triggered in SwiftUI button

### ISSUE THREAD

**Issue by kacper1703** [realm/SwiftLint#5762]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

The following code does not trigger the `no_empty_block` rule:
```swift
@ViewBuilder
private var replyButton: some View {
    Button {

    } label: {
        Text("Reply")
    }
}
```

### Environment

* SwiftLint version: 0.56.1
* Xcode version: Xcode 15.4 (15F31d)
* Installation method used: Homebrew
* Configuration file:

```yml
opt_in_rules:
- no_empty_block

no_empty_block:
  severity: error
  disabled_block_types: [initializer_bodies]
```

**Comment by SimplyDanny** [comment]
Closures are entirely ignored by the rule. That will be treated as a bug.
